meson.build

```diff
@@ -0,0 +1,376 @@
+## SPDX-License-Identifier: 0BSD
+## Author: Vincent Torri
+
+project(
+    'XZ Utils',
+    'c',
+    version: '5.6.99',
+    license: '0BSD',
+    license_files: [
+        'COPYING',
+        'COPYING.0BSD',
+        'COPYING.GPLv2',
+        'COPYING.GPLv3',
+        'COPYING.LGPLv2.1'
+    ],
+    default_options: [
+        'b_ndebug=if-release',
+        'c_std=c11',
```

Comment:
I know that some project leaders do not like to use the latest version of meson. If there is no problem for you to use meson >= 1.3, then ok.

Reply:
I actually don't know right now which Meson (or muon) version is OK to require. I just get an impression that a single value doesn't work everywhere (if MSVC needs C11 mode). I haven't checked in detail how the `c_std` option in Meson exactly works.

Reply:
https://github.com/mesonbuild/meson/blob/master/mesonbuild/compilers/c.py#L453-L485

For MSVC, meson ignores c99 and passes /std:XXX when the mode is c11 or c17. The difference is really about whether you want to opt into GNU extensions (which MSVC obviously doesn't support).

Reply:
I vaguely remember that, a very long time ago, using `gnu99` was better in practice when the intention was to use non-standard extensions (language or library) when they are available.

If `gnu99` doesn't put MSVC into C11/C17 mode then MSVC needs some other `c_std` option. MSVC is a special case in so many ways that handling it specially here could be OK too, assuming it's easy to do.

Using `gnu11` with GCC and Clang and strict C11 with compilers that don't support GNU extensions could be OK too. But if compiler doesn't support C11 then it should fallback to C99 if that is supported.

The list method that requires Meson >= 1.3.0 would make all this trivially easy. [muon](https://muon.build/releases/edge/docs/status.html) says it's currently at _meson compat version 1.1_.

So, oh well, maybe try `gnu11` for now if that helps getting MSVC to C11 mode. This is a small detail in the big picture and can be worried more a little later.

Reply:
what I did for muon with cl.exe compiler: I just pass the required standard. So, no problem with c11. (there will be a message from the compiler if the standard is not correct)
I maybe should pass the standard if it is only c11 or c17 like meson. I'll ask the author

Reply:
The Meson docs give an impression that `c_std=gnu11` would fallback to C11 with MSVC but this might become deprecated and eventually an error. So one would need to list the possible standards.

I think requiring new enough Meson and using `c_std=gnu99,c11,c99` or `c_std=gnu11,gnu99,c11,c99` is fine for now. It might not be muon compatible right now but I would expect this kind of feature to get implemented at some point in muon too. Very probably the first versions of Meson support won't be as portable as the Autotools build is and it doesn't need to be. Progress happens one step at a time.